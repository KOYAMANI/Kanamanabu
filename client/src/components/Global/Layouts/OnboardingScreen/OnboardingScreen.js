import React from 'react'
import { 
    Container, 
} from 'react-bootstrap';
import { 
    OnboardingMain,
    OnboardingTitle,
    OnboardingIntroText,
    OnboardingContainer,
    OnboardingRow
 } from './OnboardingScreenElements'

const OnboardingScreen = ({title, children}) => {
    return (
        <OnboardingMain>
        <Container>
            <OnboardingRow>
                <OnboardingIntroText>
                <OnboardingTitle>{title}</OnboardingTitle>
                    <OnboardingContainer>
                        {children}
                        
                    </OnboardingContainer>
                </OnboardingIntroText>
            </OnboardingRow> 
        </Container>
    </OnboardingMain>
       
    )
}

export default OnboardingScreen
