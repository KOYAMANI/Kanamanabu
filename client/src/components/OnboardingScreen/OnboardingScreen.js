import React from 'react'
import { 
    Container, 
} from 'react-bootstrap';
import { 
    Main,
    OnboardingTitle,
    IntroText,
    OnboardingContainer,
    OnboardingRow
 } from './OnboardingScreenElements'

const OnboardingScreen = ({title, children}) => {
    return (
        <Main>
        <Container>
            <OnboardingRow>
                <IntroText>
                <OnboardingTitle>{title}</OnboardingTitle>
                    <OnboardingContainer>
                        {children}
                        
                    </OnboardingContainer>
                </IntroText>
            </OnboardingRow> 
        </Container>
    </Main>
       
    )
}

export default OnboardingScreen
