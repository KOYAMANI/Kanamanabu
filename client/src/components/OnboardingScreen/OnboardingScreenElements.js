import styled from 'styled-components';
import background from '../../assets/sakura.png'
import { Row, } from 'react-bootstrap';

export const OnboardingMain = styled.div`
    min-height: 93vh;
    display: flex;
    align-items: center;
    background-position: center;
    background-size: cover;
    background-image: url(${background});
`

export const OnboardingIntroText = styled.div`
    width: 100%;
    text-align: center;      
`

export const OnboardingTitle = styled.h1`
    font-size: 45px;
    color: #ff7879;
    @media (max-width: 600px) {
        font-size: 25px;
        display: flex;
        justify-content: center;
      }
`

export const OnboardingRow = styled(Row)`
    background-color: #ffffff;
`

export const OnboardingSubTitle = styled.h1`
    font-size: 25px;
    padding-top: 10px;
`

export const OnboardingContainer = styled.div`
    margin-top: 50px;
    display: flex;
    justify-content: space-evenly;
`