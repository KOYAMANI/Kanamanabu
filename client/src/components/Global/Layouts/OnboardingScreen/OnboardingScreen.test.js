import {render, screen, cleanup} from '@testing-library/react';
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer'
import OnboardingScreen from "./OnboardingScreen";

afterEach(cleanup)

describe('ErrorMessage', () =>{
    test('Render without crashing', ()=>{
        const div = document.createElement('div')
        ReactDOM.render(<OnboardingScreen title="onboarding-test" >onboarding-child</OnboardingScreen>, div)
    });

    test('Render text', ()=>{
        render(<OnboardingScreen title="onboarding-test" >onboarding-child</OnboardingScreen>)
        const element = screen.getByText('onboarding-test')
        const child = screen.getByText('onboarding-child')
        expect(element).toBeInTheDocument();
        expect(element).toHaveTextContent('onboarding-test')
        expect(child).toHaveTextContent('onboarding-child')
    })

    test('match snapshot',()=>{
        const tree = renderer.create(<OnboardingScreen title="onboarding-test" >onboarding-child</OnboardingScreen>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})