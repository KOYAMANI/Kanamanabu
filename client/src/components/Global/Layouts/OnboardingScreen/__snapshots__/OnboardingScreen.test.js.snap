// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage match snapshot 1`] = `
<div
  className="sc-bczRLJ iWdtwL"
>
  <div
    className="container"
  >
    <div
      className="sc-hKMtZM nVtxo row"
    >
      <div
        className="sc-gsnTZi bgSfay"
      >
        <h1
          className="sc-dkzDqf dNCfZL"
        >
          onboarding-test
        </h1>
        <div
          className="sc-jSMfEi RDsOs"
        >
          onboarding-child
        </div>
      </div>
    </div>
  </div>
</div>
`;
